### ISSUE TITLE
xsl:if variable comparison test not compiled correctly

### ISSUE THREAD

**Issue by tobyzerner** [s9e/TextFormatter#125]
I have a tag template:

```xml
<a href="{@url}">
  <xsl:attribute name="class">
    foo
    <xsl:if test="@id = $USER_ID">self</xsl:if>
  </xsl:attribute>
  <xsl:value-of select="@name"/>
</a>
```

But it's not working! It outputs the `self` class no matter what the `@id` is; the if condition always evaluates to true.

I am using the PHP rendering engine – this problem doesn't occur without it. Upon inspecting the cached Renderer_xxxxx.php class, I can see that that's because it's not actually compiling the full condition. The relevant part is this:

```php
$this->out.='<a href="'.htmlspecialchars($node->getAttribute('url'),2).'" class="foo ';if($node->getAttribute('id'))$this->out.='self';$this->out.='">'.htmlspecialchars($node->getAttribute('name'),0).'</a>';
```

or, more specifically:

```php
if($node->getAttribute('id'))$this->out.='self';
```

**Comment by JoshyPHP** [comment]
What version of the library do you use? Here's what I get on 2.1.2:

```php
$this->out.='<a href="'.htmlspecialchars($node->getAttribute('url'),2).'" class="
    foo
    ';if($node->getAttribute('id')===$this->params['USER_ID'])$this->out.='self';$this->out.='">'.htmlspecialchars($node->getAttribute('name'),0).'</a>';
```

**Comment by tobyzerner** [comment]
Doh, my composer.json was still stuck on version 1. Thanks @JoshyPHP!

**Comment by JoshyPHP** [comment]
It's interesting because even on 1.x that's not something that should really happen. I can reproduce it locally on PHP 7.3, although the same tests produce a different result on Travis's PHP 7.3 instances.

I tracked it down to a change in behaviour in the way captures in handled in subroutines. It's hard to reproduce because it requires a specific nesting of some capturing subpatterns so my guess it that's a bug in PCRE2 itself, although it may require a specific environment or compilation tools to trigger. Here's the output from pcretest vs pcre2test:
```
$ pcretest 
PCRE version 8.43 2019-02-23

  re> /(*F)((x))|((?1))/
data> x
 0: x
 1: <unset>
 2: <unset>
 3: x
data> 
$ pcre2test 
PCRE2 version 10.33 2019-04-16
  re> /(*F)((x))|((?1))/
data> x
 0: x
 1: <unset>
 2: x
 3: x
data> 
```

Version >=2.0 of the library isn't affected because the XPath-to-PHP convertor has been rewritten and it uses `(*MARK)` to identify which branch of the regexp matched rather than the content of the captures. I may fix it in the 1.4 branch for completeness but I'm not sure I'll make another 1.4 release so upgrading to the most recent release is the best way to avoid that issue.
